### ISSUE TITLE
get_version missing

### ISSUE THREAD

**Issue by oddjobz** [twisted/txmongo#186]
Hi, the newer versions of pymongo have a new gridfs call "get_version" to recover all or specific versions of any given file in "gridfs". This is actually pretty critical if you're using the versioning features of Mongo .. I've had a go at implementing this, seems to work for me, but if we could have a "get_version" call, either this code or something better, that would be great;

```python
@defer.inlineCallbacks
def get_version(self, filename=None, version=-1, **kwargs):
    if version < 0:
        skip = abs(version) 
        self.__files.ensure_index(filter.sort(ASCENDING("filename") + DESCENDING("uploadDate")))
        cursor = yield self.__files.find({"filename": filename}, filter=filter.sort(DESCENDING("uploadDate")),skip=skip,limit=1)
    else:
        self.__files.ensure_index(filter.sort(ASCENDING("filename") + ASCENDING("uploadDate")))
        cursor = yield self.__files.find({"filename": filename}, filter=filter.sort(ASCENDING("uploadDate")),limit=1,skip=skip)
    try:
        defer.returnValue(GridOut(self.__collection, cursor[0]))
    except StopIteration:
        raise NoFile("no version %d for filename %r" % (version, filename))

```

**Comment by IlyaSkriblovsky** [comment]
Thanks for suggestion. It would be cool if you can send us a pull request!

**Comment by oddjobz** [comment]
Mmm, I'm not sure how I submit a pull request without access to the repo? (I have no access to create or update a branch...)

**Comment by IlyaSkriblovsky** [comment]
You can clone repo into your own account, create branch with proposed changes there and then click "New pull request" button. This will open a pull request from your branch in *your repo* into master branch in *txmongo repo*.

Github would be a very dull place if nobody can create pull requests without write permission :)

By the way, as far as I can see, PyMongo 3.1 removed `ensure_index` calls from `get_version`. I think it is good because implicit index creation can be unsafe in production.

**Comment by psi29a** [comment]
At the top right click on 'fork', it will create a fork of this repo as your repo. Make a branch on your repo, commit changes, then when ready a green button should be available, something like make or pepare merge (or pull) request.

**Comment by psi29a** [comment]
Since this is your commit, you'll get attribution. :)

Some things to add:
* changelog entry for next release
* add yourself to the list of contributors

**Comment by oddjobz** [comment]
Ok, I've generated my first "pull request" .. if I've done something wrong, please don't shoot me .. :)

**Comment by IlyaSkriblovsky** [comment]
Closed by #189
